src/main/java/org/opentripplanner/middleware/controllers/api/ApiController.java

```diff
@@ -140,18 +145,35 @@ public void bind(final SparkSwagger restApi) {
     }
 
     /**
-     * HTTP endpoint to get multiple entities.
+     * HTTP endpoint to get multiple entities based on the user permissions
      */
+    // FIXME Maybe better if the user check (and filtering) was done in a pre hook?
+    // FIXME Will require further granularity for admin
     private List<T> getMany(Request req, Response res) {
-        return persistence.getAll();
+
+        Auth0UserProfile requestingUser = getUserFromRequest(req);
+        if (isUserAdmin(requestingUser)) {
+            return persistence.getAll();
+        } else if (persistence.clazz == OtpUser.class) { 
+            Bson filter = Filters.eq("_id", requestingUser.otpUser.id);
+            return persistence.getFiltered(filter);
+        } else {
+            Bson filter = Filters.eq("userId", requestingUser.otpUser.id);
+            return persistence.getFiltered(filter);
+        }
```

Comment:
I would like to see some comments accompanying this block of conditionals describing the intent. It may be obvious to all of us at this moment, but future readers may not fully understand the context for the logic we've embedded here. E.g., "If user is an admin, the context is presumed to be the admin dashboard, so we deliver all entities (e.g., monitored trips) for management or review" etc...

Reply:
updated